src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs

```diff
@@ -272,7 +272,11 @@ private sealed class DebugFinalizableAsyncStateMachineBox<TStateMachine> : // SO
         /// <summary>A strongly-typed box for Task-based async state machines.</summary>
         /// <typeparam name="TStateMachine">Specifies the type of the state machine.</typeparam>
         [DebuggerDisplay("{DebuggerDisplay,nq}")]
-        private class AsyncStateMachineBox<TStateMachine> : // SOS DumpAsync command depends on this name
+        private
+#if NATIVEAOT
+        sealed
+#endif
```

Comment:
Can we disable this warning in NativeAot's corelib? If we have to ifdef around it, or add pragma suppressions, the analyzer becomes a net negative contributor: native AOT doesn't need `sealed`, [the AOT compiler will place `sealed` on things automatically](https://migeel.sk/blog/2023/11/22/top-3-whole-program-optimizations-for-aot-in-net-8/) based on whole program analysis and it does a better job than this Roslyn analyzer.

If we have to do extra work to accommodate the Roslyn analyzer, we should just turn it off for the whole assembly. It will not have a positive contribution and it's already a negative.

Reply:
Sure we can just disable CA1852 again, I only enabled it because it was in the `<!-- we should just fix -->` NoWarn group.

Reply:
@MichalStrehovsky updated